web/src/features/charts/ProductionSourceLegendList.tsx

```diff
@@ -1,26 +1,42 @@
+import { Button } from 'components/Button';
 import { twMerge } from 'tailwind-merge';
 import { ElectricityModeType } from 'types';
 
+import { SelectedData } from './OriginChart';
 import ProductionSourceLegend from './ProductionSourceLegend';
 
 export default function ProductionSourceLegendList({
   sources,
   className,
+  selectedData,
+  isDataInteractive = false,
 }: {
   sources: ElectricityModeType[];
   className?: string;
+  selectedData?: SelectedData;
+  isDataInteractive?: boolean;
 }) {
+  // TODO(cady): memoize
   return (
-    <div className={twMerge('flex w-fit flex-row flex-wrap gap-2.5 py-1', className)}>
+    <div className={twMerge('flex w-fit flex-row flex-wrap gap-1 py-1', className)}>
       {sources.map((source, index) => {
+        const onClick = () => selectedData?.toggle(source);
         const capitalizedLabel = `${source.charAt(0).toUpperCase()}${source.slice(1)}`;
+        const isSourceSelected = selectedData?.isSelected(source);
+
         return (
-          <span key={source} className="flex items-center gap-1 text-sm">
-            <ProductionSourceLegend key={index} electricityType={source} />
-            <p className="text-xs text-neutral-600 dark:text-gray-300">
-              {capitalizedLabel}
-            </p>
-          </span>
+          <Button
+            isDisabled={!isDataInteractive}
+            key={index}
+            type="tertiary"
+            size="sm"
+            foregroundClasses={'text-xs font-normal text-neutral-600 dark:text-gray-300'}
+            backgroundClasses={isSourceSelected ? 'bg-neutral-400/10' : ''}
+            onClick={onClick}
+            icon={<ProductionSourceLegend key={index} electricityType={source} />}
+          >
+            {capitalizedLabel}
+          </Button>
```

Comment:
It's kind of hard to see which are selected in based on the buttons alone.

I would consider adding the success green as an outline to the button if it's selected.

Example:
<img width="437" alt="image" src="https://github.com/user-attachments/assets/eb9e61c1-590d-483c-8129-e52f53ff9025">

Solar and hydro are selected here but the contrast/difference between them are too low. Especially for someone that might not have perfect eye sight.

Reply:
That's a fair point. I agree, that is difficult to see. However I think we use a green outline to indicate that an element is focused, so I think we should use a different outline color instead. Here's my update:

<img width="361" alt="Screenshot 2024-11-14 at 15 39 11" src="https://github.com/user-attachments/assets/c7385190-206d-4d9e-80d7-08d9d55ae4d6">
<img width="357" alt="Screenshot 2024-11-14 at 15 42 05" src="https://github.com/user-attachments/assets/161886d9-215b-4916-bf99-0861ad17be48">

For dark mode, I used the bg and outline colors of the TimeController toggle and added a darker outline for the light theme.

Reply:
Not using green is a fair point, but this already looks a lot better!

I'd keep @Alportan in the loop but I don't think he'll have any objections to this approach, but maybe a suggestion or two.

Otherwise this LGTM from my side of things and I'll approve it but maybe wait just a tiny bit to merge it if Alex has any feedback?

Reply:
Looks good to me! 🙌 

Nitpick: We can give more contrast to the selected label text, but I think it's already clear enough which sources are selected now 💪

Reply:
@Alportan what did you have in mind? Bold text? Black text/white text depending on theme?

Reply:
More like black/white text, to make it pop 🎈

Reply:
Better?
<img width="427" alt="Screenshot 2024-11-15 at 10 55 17" src="https://github.com/user-attachments/assets/50692353-da8f-4656-84ed-fcd3087fc9b5">
<img width="423" alt="Screenshot 2024-11-15 at 10 56 37" src="https://github.com/user-attachments/assets/bfa660ee-4d86-4565-b9b5-74e2e155fe5d">

Reply:
Yeees 🤩 Very small change, but it already stand out a bit more 🙌